modules/caddyhttp/rewrite/caddyfile.go

```diff
@@ -213,6 +213,9 @@ func applyQueryOps(h httpcaddyfile.Helper, qo *queryOps, args []string) error {
 		renameValKey := strings.Split(key, ">")
 		qo.Rename = append(qo.Rename, queryOpsArguments{Key: renameValKey[0], Val: renameValKey[1]})
 
+	case len(args) == 3:
+		qo.Replace = append(qo.Replace, queryOpsReplacement{Key: key, Search: args[1], Replace: args[2]})
```

Comment:
I don't remember how we did it for header replacement, did we have `SearchRegexp` support from Caddyfile? If not, I think what we should do is allow `~` as a prefix to the search string, and if so it becomes a regexp. That's the way we did it for the `map` directive: https://caddyserver.com/docs/caddyfile/directives/map#syntax. Maybe we can apply that to the header replacements Caddyfile parsing too.

Reply:
I thin the header replacement uses the `SearchRegexp` exclusively when it come to the Caddyfile support. Do we want to go with that or use the `~` prefix?

Reply:
Oh :thinking: I guess that's true. Maybe we should do the same here then and only do regexp. WDYT @mholt

Reply:
Hmm, I don't remember why we did the header Caddyfile that way; but we could do the same here I suppose. Regex is more powerful :man_shrugging: